tax-processing-functions/quality_and_split/main.py

```diff
@@ -0,0 +1,403 @@
+    result_ents = []
+    for entity in extracted_doc.entities:
+        entity_type = str(entity.type_)
+        if entity_type == "line_item":
+            for p in entity.properties:
+                ents = {
+                    "type_": p.type_,
+                    "mention_text": p.mention_text,
+                    "confidence": p.confidence,
+                }
+                result_ents.append(ents)
+            ents = {
+                "type_": entity.type_,
+                "mention_text": entity.mention_text,
+                "confidence": entity.confidence,
+            }
+            result_ents.append(ents)
+        else:
+            ents = {
+                "type_": entity_type,
+                "mention_text": entity.mention_text,
+                "confidence": entity.confidence,
+            }
+            result_ents.append(ents)
+    print("Formatted entities to a list.")
+    return result_ents
+
+
+# Write processed JSON results to BigQuery table
+def write_to_bq(
+    project_name, dataset_name, table_name, extracted_list, file_name, type
+):
+    # project_name = string, Google Cloud Project name
+    # dataset_name = string, BQ dataset name to import to
+    # table_name = string, BQ table name to import to
+    # extracted_list = data in JSON list format
+    # file_name = document name
+    # type = document type
+    table_ref = bigquery.dataset.DatasetReference(project_name, dataset_name).table(
+        table_name
+    )
+    date_loaded = datetime.now().strftime(
+        ts_format
+    )  # Create timestamp processed_date in string format
+
+    for item in extracted_list:
+        item["doc_name"] = file_name
+        item["doc_type"] = type
+        item["processed_date"] = date_loaded
+
+    json_data = json.dumps(extracted_list, sort_keys=False)
+
+    # Convert to a JSON Object
+    json_object = json.loads(json_data)
+
+    # BQ Schema
+    job_config = bigquery.LoadJobConfig(
+        source_format=bigquery.SourceFormat.NEWLINE_DELIMITED_JSON,
+        ignore_unknown_values=True,
+        write_disposition="WRITE_APPEND",
+        schema=[
+            bigquery.SchemaField("processed_date", "STRING"),
+            bigquery.SchemaField("doc_name", "STRING"),
+            bigquery.SchemaField("doc_type", "STRING"),
+            bigquery.SchemaField("confidence", "FLOAT"),
+            bigquery.SchemaField("type_", "STRING"),
+            bigquery.SchemaField("mention_text", "STRING"),
+        ],
+    )
+    job = bq_client.load_table_from_json(json_object, table_ref, job_config=job_config)
+    print(f"BQ upload job results:{job.result()}")
+
+
+# Upload files to GCP Bucket
+def upload_blob(bucket_name, file_name, json_string):
+    # bucket_name = "your-bucket-destination-name" in GCS
+    # file_name = intended file name "local/path/to/file/destination"
+    # json_string = data to be moved to GCS bucket in json string format
+    nd_output = ndjson.dumps(json.loads(json_string))
```

Comment:
What is the JSON string you're uploading?
You might be able to avoid using the ndjson library depending on what format the json is for.

Reply:
the json string is the extracted and processed document in json format